mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ExecutorStateManagerImpl.java

```diff
@@ -372,12 +372,23 @@ private Optional<Map<TaskExecutorID, TaskExecutorState>> findBestFitFor(TaskExec
      * @return true if all allocation constraints are satisfied, false otherwise.
      */
     public boolean areSchedulingAttributeConstraintsSatisfied(SchedulingConstraints constraints, Map<String, String> teAssignmentAttributes) {
+        Map<String, String> teAssignmentAttributesLowercased = teAssignmentAttributes.entrySet()
+            .stream()
+            .collect(Collectors.toMap(entry -> entry.getKey().toLowerCase(), Map.Entry::getValue));
+
+        Map<String, String> constraintsAttributesLowercased = constraints.getSchedulingAttributes().entrySet()
+            .stream()
+            .collect(Collectors.toMap(entry -> entry.getKey().toLowerCase(), Map.Entry::getValue));
```

Comment:
Can these manipulations done instead at the place where these attributes are being created? Also, what's the need for getting rid of the case?

Reply:
Maintaining conventions like low case throughout the codebase can be tough tbh, that's why to avoid any possible mismatch and scheduling issues I checked it in here. we can always revisit this I guess. What would be the benefits of a case-sensitive approach in this case?